Assets/_System/Scripts/TurnBasedScripts/CombatManager.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CombatManager : MonoBehaviour
+{
+    [Header("Selected Unit")]
+    public GridObject selectedUnit = null;   // Currently highlighted GridTile
+
+    [Header("Players")]
+    public CombatPlayer gamePlayer = new CombatPlayer("Player");
+    public CombatPlayer enemyPlayer = new CombatPlayer("Enemy");
+
+    public CombatPlayer currentPlayer;
+    public CombatPlayer otherPlayer;
+
+    public static CombatManager Instance = null;
+
+    protected void Awake()
+    {
+        if (Instance == null)
```

Comment:
It would be better to make this a Service. Services are meant to be a better alternative to singletons.

Reply:
Made a service.